src/main/java/io/sniffy/Sniffer.java

```diff
@@ -97,6 +113,113 @@ public static int executedStatements() {
+     * @param expectationList a list of {@link Expectation} annotations
+     * @return a new {@link Spy} instance with given expectations
+     * @see #spy()
+     * @since 2.1
+     */
+    public static Spy expect(List<Expectation> expectationList) {
+        Spy spy = Sniffer.spy();
+
+        for (Expectation expectation : expectationList) {
+
+            Range range = Range.parse(expectation);
+
+            if (-1 != range.value) {
+                spy.expect(SqlQueries.exact(range.value).threads(expectation.threads()).type(expectation.query()));
+            } else if (-1 != range.min && -1 != range.max) {
+                spy.expect(SqlQueries.between(range.min, range.max).threads(expectation.threads()).type(expectation.query()));
+            } else if (-1 != range.min) {
+                spy.expect(SqlQueries.min(range.min).threads(expectation.threads()).type(expectation.query()));
+            } else if (-1 != range.max) {
+                spy.expect(SqlQueries.max(range.max).threads(expectation.threads()).type(expectation.query()));
+            }
+        }
+
+        return spy;
+    }
+
+    /**
+     * Executable interface is similar to {@link java.lang.Runnable} but it allows throwing {@link java.lang.Exception}
+     * from it's {@link #execute()} method
+     */
+    public interface Executable {
+
+        /**
+         * When {@link Sniffer#execute(Sniffer.Executable)}
+         * method is called, it will execute the Executable.execute() method, record the SQL queries and return the
+         * {@link Spy} object with stats
+         * @throws Exception code under test can throw any exception
+         */
+        void execute() throws Throwable;
+
+    }
+
+    /**
+     * Execute the {@link Sniffer.Executable#execute()} method, record the SQL queries
+     * and return the {@link Spy} object with stats
+     * @param executable code to test
+     * @return statistics on executed queries
+     * @throws RuntimeException if underlying code under test throws an Exception
+     */
+    public static Spy execute(Executable executable) {
+        return spy().execute(executable);
+    }
+
+    /**
+     * Execute the {@link Runnable#run()} method, record the SQL queries
+     * and return the {@link Spy} object with stats
+     * @param runnable code to test
+     * @return statistics on executed queries
+     */
+    public static Spy run(Runnable runnable) {
+        return spy().run(runnable);
+    }
+
+    /**
+     * Execute the {@link Callable#call()} method, record the SQL queries
+     * and return the {@link SpyWithValue} object with stats
+     * @param callable code to test
+     * @param <V> type of return value
+     * @return statistics on executed queries
+     * @throws Exception if underlying code under test throws an Exception
+     */
+    @SuppressWarnings("unchecked")
+    public static <V> SpyWithValue<V> call(Callable<V> callable) throws Exception {
+        return spy().call(callable);
+    }
+
```

Comment:
![Codacy](https://www.codacy.com/assets/images/favicon.png) Issue found: [Fields should be declared at the top of the class, before any method declarations, constructors, initializers or inner classes.](https://www.codacy.com/app/sniffy/sniffy/file/2495663886/issues/source?bid=3005960&fileBranchId=3034483#l205)